Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py

```diff
@@ -20,22 +20,64 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import itertools
 import json
 import sys
-from re import search
+import re
 
 from .command import Command
 from .find import Info
 from .trace import Trace, Relationship, CommitsStory
 from datetime import datetime
-from webkitcorepy import arguments
+from webkitcorepy import arguments, run, string_utils
 from webkitscmpy import Commit, local
 
 
+def fuzzy_filter(string):
+    return re.compile(string)
+
+
+if sys.version_info > (3, 6):
+    try:
+        from rapidfuzz import fuzz
+
+        def fuzzy_filter(string, ratio=95):
+            return lambda content: fuzz.partial_ratio(string, content) > ratio
+    except ModuleNotFoundError:
+        pass
+
+
 class Pickable(Command):
     name = 'pickable'
     help = 'List commits in a range which can be cherry-picked'
 
+    DEFAULT_PATTERNS = [
+        'Versioning',
+        'Gardening',
+        'Build-Fix',
+        'Apply-Patch',
+    ]
+    PATTERNS = {
+        'Cherry-pick': [re.compile(r'[Cc]herry[- ][Pp]ick')],
+        'Versioning': [re.compile(r'[Vv]ersioning')],
```

Comment:
I think these might be too broad to be allowed to match anywhere in the commit message first line.

Examples of would-be false negatives for "cherry-pick": 4f144420a531751b9d7f43a741713ae5ca330b24 8de977259722
and for "versioning": 690a2333ae979af678f1d910cf455a129757f2ff f3e5aebe67ea0146cb14da9cd46e5e5db9410a10

Reply:
The `cherry-pick` example is disabled by default in favor of a more narrow one.

Let me see if I can narrow "versioning" any further....

Reply:
We can! Narrowed the `Versioning` check, along with a few others, mostly by tying them to the start of strings.